### PR TITLE
Implementing google analytics for "v1.5" release features

### PR DESCRIPTION
**Pull request by fertandil87**
https://globalfishingwatch.atlassian.net/browse/MAP-666

### CODE REVIEW

File: apps/fishing-map/features/analysis/Analysis.tsx [hunk 1]
```diff
@@ -161,10 +164,32 @@ function Analysis() {
...
+          category: 'Analysis',
+          action: `Click '${option.title}' in analysis mode`,
+          label: JSON.stringify({
+            regionName: analysisAreaName,
+            sourceNames: dataviews.flatMap(dataview => getSourcesSelectedInDataview(dataview).map(source => source.label))
+          })
+        })
+      }
+      if (option.id === 'periodComparison' && timeComparison) {
+        console.log(timeComparison)
```
Comment (j8seangel) [inline apps/fishing-map/features/analysis/Analysis.tsx:178]: 👀

File: apps/fishing-map/features/analysis/Analysis.tsx [hunk 2]
```diff
@@ -161,10 +164,32 @@ function Analysis() {
...
+        uaEvent({
+          category: 'Analysis',
+          action: `Click '${option.title}' in analysis mode`,
+          label: JSON.stringify({
+            duration: timeComparison.duration + ' ' + timeComparison.durationType,
```
Comment (j8seangel) [inline apps/fishing-map/features/analysis/Analysis.tsx:183]: would timeComparison have the updated value at this point?

File: apps/fishing-map/features/analysis/Analysis.tsx (hunk 1 above)
Comment (fertandil87) [reply apps/fishing-map/features/analysis/Analysis.tsx:178]: ups I removed logs but I miss this one

File: apps/fishing-map/features/analysis/Analysis.tsx (hunk 2 above)
Comment (fertandil87) [reply apps/fishing-map/features/analysis/Analysis.tsx:183]: I don't know a reason to don't have it updated, this is fired when we change tabs

File: apps/fishing-map/features/analysis/AnalysisPeriodComparison.tsx [hunk 3]
```diff
@@ -37,6 +39,58 @@ const AnalysisPeriodComparison: React.FC<AnalysisTypeProps> = (props) => {
...
+    uaEvent({
+      category: 'Analysis',
+      action: `Select baseline date in 'period comparison'`,
+      label: JSON.stringify({
+        date: date.target.value,
+        regionName: analysisAreaName,
+        sourceNames: dataviews.flatMap(dataview => getSourcesSelectedInDataview(dataview).map(source => source.label))
+      })
+    })
+    onCompareStartChange(date)
```
Comment (rdgfuentes) [inline apps/fishing-map/features/analysis/AnalysisPeriodComparison.tsx:65]: I think you should be calling `onStartChange` here, look at this line: https://github.com/GlobalFishingWatch/frontend/pull/1204/files#diff-63134b5729019ef9b974a721d2a391154fc931c4210efad2c398264111284229L101

**Pull request merged**